### PR TITLE
feat: add track details component with additional details

### PR DESCRIPTION
**Pull request by rajat-ws**
## Description:

- Added `Show Details` button
- On clicking the ShowDetails button, additional details are displayed along in the `TrackCard`

## GIF :

q

### CODE REVIEW

File: features/itunes/components/TrackCard/index.tsx [hunk 1]
```diff
@@ -1,16 +1,18 @@
-import { isEmpty } from "lodash";
+import { useRouter } from "next/router";
 import styled from "styled-components";
+import { isEmpty } from "lodash";
```
Comment (ihatelactose) [inline features/itunes/components/TrackCard/index.tsx]: Generally when using lodash in frontend it is recommended to destructure it saves a lot of initial load time and space.

```suggestion
import isEmpty from "lodash/isEmpty";
```

Comment (rajat-ws) [reply features/itunes/components/TrackCard/index.tsx]: Oh, thanks for the update. Will update it accordingly

**Comment by ihatelactose** [comment]
Just fix the thing above and this is good to! An amazing pr just as I expected!

**Review by ihatelactose — APPROVED** [review]

File: features/itunes/components/TrackCard/index.tsx (hunk 1 above)
Comment (rajat-ws) [reply features/itunes/components/TrackCard/index.tsx]: Updated accordingly in the latest commit.

**Comment by rajat-ws** [comment]
Merging it now. Thanks for the review and approval @ihatelactose

**Pull request merged**